### PR TITLE
Add warning if the GPU function is called and CUDA is not available

### PR DESCRIPTION
**Pull request by logankilpatrick**
This should close https://github.com/FluxML/Flux.jl/issues/1634. Thanks @mcabbott for the help : )

### CODE REVIEW

**Comment by logankilpatrick** [comment]
Sample output: 
```julia
julia> using Flux
[ Info: Precompiling Flux [587475ba-b771-5e3f-ad9e-33799f191a9c]

julia> m = Dense(10,5) |> gpu
┌ Warning: The GPU function is being called despite the GPU not being accessible. Defaulting back to the CPU.
└ @ Flux ~/Desktop/Flux.jl/src/functor.jl:187
Dense(10, 5)        # 55 parameters
```

**Review by DhairyaLGandhi — COMMENTED** [review]
We typically encourage users to use the `gpu` function if they want to make their code portable both with and without GPUs. We keep the overhead of calling `gpu` low so that it doesn't hurt the performance but keeps code flexible. I'm worried this would discourage users from using `gpu` properly.

**Comment by logankilpatrick** [comment]
Maybe @info would be better? I don't think it's a huge deal but it is rather unintuitive to me to call a function which doesn't actually do what I think it will do (move stuff to gpu).

File: src/functor.jl [hunk 1]
```diff
@@ -183,6 +183,9 @@ function check_use_cuda()
     if use_cuda[] && !CUDA.has_cudnn()
       @warn "CUDA.jl found cuda, but did not find libcudnn. Some functionality will not be available."
     end
+    if use_cuda[] == false
```
Comment (DhairyaLGandhi) [inline src/functor.jl:186]:
```suggestion
    if !(use_cuda[])
```

Reads cleaner

**Comment by logankilpatrick** [comment]
```Julia
julia> m = Dense(10,5) |> gpu
[ Info: The GPU function is being called despite the GPU not being accessible. Defaulting back to the CPU.
Dense(10, 5)        # 55 parameters
```
This seems a bit less aggressive?

File: src/functor.jl [hunk 2]
```diff
@@ -183,6 +183,9 @@ function check_use_cuda()
     if use_cuda[] && !CUDA.has_cudnn()
       @warn "CUDA.jl found cuda, but did not find libcudnn. Some functionality will not be available."
     end
+    if !(use_cuda[])
+      @warn "The GPU function is being called despite the GPU not being accessible. Defaulting back to the CPU." maxlog=1
```
Comment (darsnack) [inline src/functor.jl:187]:
```suggestion
      @warn """The GPU function is being called but the GPU is not accessible. 
               Defaulting back to the CPU. (No action is required if you want to run on the CPU).""" maxlog=1
```

**Review by darsnack — COMMENTED** [review]
I think it is better to inform the user when we silently do something unexpected, so I like this change. It only prints out once, so with the right language, I think should not discourage usage of `gpu`.

**Review by darsnack — APPROVED** [review]
LGTM!

**Comment by ToucheSir** [comment]
bors try

**Comment by ToucheSir** [comment]
(doing a bit of an experiment here to see if we can use `try` (which appears to work all the time) over `r+` (which has been consistently failing recently for some reason, along with master) until the whole Buildkite complaining on certain branches only thing is figured out).

**Review by ToucheSir — APPROVED** [review]
bors r+

**Comment by ToucheSir** [comment]
Ok, everything but that one sometimes successful, sometimes broken ConvTranspose test works on integration. I don't think that should hold up a change as innocuous as this, so merging manually.

**Pull request merged**